### PR TITLE
feat: DIA-1511: Support data batch context for autorefinement

### PR DESCRIPTION
**Pull request by matt-bernstein**
fix /improve-prompt request to not require a jobid, as a job has not yet been created

### CODE REVIEW

**Comment by niklub** [comment]
Does this PR introduce any functional changes or is it only about reformatting and renaming?

**Comment by matt-bernstein** [comment]
the only functional change is this
> fix /improve-prompt request to not require a jobid, as a job has not yet been created

which I assume was a bug, unless it's to support some future pathway what I didn't know about

**Review by pakelley — APPROVED** [review]

**Comment by pakelley** [comment]
Approved as far as these changes being fine, but to be clear: do we just not actually need any substantive changes (aside from not requiring the job_id) to support this?

**Comment by matt-bernstein** [comment]
> Approved as far as these changes being fine, but to be clear: do we just not actually need any substantive changes (aside from not requiring the job_id) to support this?

That's right. It's a bit hard to follow all the plumbing, but the PromptImprovementSkill is currently implemented as an AnalysisSkill: DataFrame -> Record, where all other inputs go in the static `instructions` field and the GT data batch is the DataFrame. This needs some rethinking in the future if we want the ability to input multiple kinds of example data, such as with and without GT, or positive (correct) vs negative (incorrect) examples. Especially since we don't actually want anything implemented in the AnalysisSkill (just batching, afaik), But that's what we have for now.

**Comment by matt-bernstein** [comment]
> Approved as far as these changes being fine, but to be clear: do we just not actually need any substantive changes (aside from not requiring the job_id) to support this?

That's right. It's a bit hard to follow all the plumbing, but the PromptImprovementSkill is currently implemented as an AnalysisSkill: DataFrame -> Record, where all other inputs go in the static `instructions` field and the GT data batch is the DataFrame. This needs some rethinking in the future if we want the ability to input multiple kinds of example data, such as with and without GT, or positive (correct) vs negative (incorrect) examples. Especially since we don't actually want anything implemented in the AnalysisSkill (just batching, afaik), But that's what we have for now.

**Review by niklub — APPROVED** [review]

**Review by hakan458 — APPROVED** [review]
